JSDemos/Demos/TabPanel/Overview/Angular/app/app.component.html

```diff
@@ -1,64 +1,84 @@
-    <div class="tabpanel-item">
-      <div>
-        <p>
-          <b>{{ company.City }}</b>
-          (<span>{{ company.State }}</span
-          >)
-        </p>
-        <p>
-          <span>{{ company.Zipcode }}</span>
-          <span>{{ company.Address }}</span>
-        </p>
-      </div>
-      <div>
-        <p>
-          Phone: <b>{{ company.Phone }}</b>
-        </p>
-        <p>
-          Fax: <b>{{ company.Fax }}</b>
-        </p>
-        <p>
-          Website:
-          <a href="{{ company.Website }}" target="_blank">
-            {{ company.Website }}
-          </a>
-        </p>
+<div class="tabpanel-demo">
+  <div class="widget-container">
+    <dx-tab-panel
+      class="dx-theme-background-color"
+      width="100%"
+      [height]="418"
+      [animationEnabled]="true"
+      [swipeEnabled]="true"
+      [dataSource]="dataSource"
+      [selectedIndex]="selectedIndex"
+      [tabsPosition]="tabsPosition"
+      [stylingMode]="stylingMode"
+      [iconPosition]="iconPosition"
+      [showNavButtons]="showNavButtons"
+    >
+      <div *dxTemplate="let tabPanelItem of 'item'">
+        <div class="tabpanel-item">
+          <div
+            *ngFor="let task of tabPanelItem.tasks"
+            [className]="getTaskClassName(task.priority)"
+          >
+            <p class="task-item-text">
+              {{ task.text }}
+            </p>
+
+            <p class="task-item-info">
+              {{ task.date }} by {{ task.assignedBy }}
+            </p>
+
+            <div class="task-item-pseudo-button-wrapper">
+              <i class="task-item-pseudo-button dx-icon dx-icon-overflow"></i>
+            </div>
+          </div>
+        </div>
       </div>
-    </div>
-  </div>
-</dx-tab-panel>
-<div class="item-box">
-  Item <span>{{ tabPanel.selectedIndex + 1 }}</span> of
-  <span>{{ itemCount }}</span>
-</div>
-<div class="options">
-  <div class="caption">Options</div>
-  <div class="option">
-    <dx-check-box text="Loop enabled" [(value)]="tabPanel.loop"></dx-check-box>
+    </dx-tab-panel>
   </div>
-  <div class="option">
-    <dx-check-box
-      text="Animation enabled"
-      [(value)]="tabPanel.animationEnabled"
-    ></dx-check-box>
-  </div>
-  <div class="option">
-    <dx-check-box
-      text="Swipe enabled"
-      [(value)]="tabPanel.swipeEnabled"
-    ></dx-check-box>
+
+  <div class="options">
+    <div class="caption">Options</div>
+
+    <div class="option">
+      <div class="option-label">Tabs position</div>
+
+      <dx-select-box
+        [inputAttr]="{ 'aria-label': 'Tabs positions' }"
+        [items]="tabsPositions"
+        [value]="tabsPosition"
+        (onValueChanged)="(onTabsPositionChanged)"
+      ></dx-select-box>
+    </div>
+
+    <div class="option">
+      <div class="option-label">Styling mode</div>
+
+      <dx-select-box
+        [inputAttr]="{ 'aria-label': 'Tabs positions' }"
+        [items]="stylingModes"
+        [value]="stylingMode"
```

Comment:
Why we should not use two way binding here - [(value)]. Seems we can.

Reply:
For what?

Reply:
It allows us not declare addition handler onValueChanged, and this is a more convenient syntax to achieve the required functionality.

Reply:
[9314317e2a169f6a48660ae0ab31835edd5370e8]